lib/json-schema/util/uri.rb

```diff
@@ -23,6 +23,46 @@ def self.normalized_uri(uri, base_path = Dir.pwd)
         normalized_uri
       end
 
+      def self.absolutize_ref(ref, base)
+        ref_uri = strip_fragment(ref.dup)
+
+        if ref_uri.absolute?
```

Comment:
The [style guide](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals) suggests using guard clauses as in the refactored code. I personally prefer them (in our team we decided to usually follow the style guide).
But that's no religious comment, I just want to put that on the table.